ui/lib/feed/src/components/editable-reflection/editable-reflection-resolver.tsx

```diff
@@ -1,50 +1,54 @@
 import * as React from 'react';
 import EditableReflection from './index';
 import { hasOwn, mapReflectEntryData, useRootComponentProps } from '@akashaorg/ui-awf-hooks';
-import { useGetReflectionByIdSuspenseQuery } from '@akashaorg/ui-awf-hooks/lib/generated/apollo';
+import { useGetReflectionByIdQuery } from '@akashaorg/ui-awf-hooks/lib/generated/apollo';
+import EntryCardLoading from '@akashaorg/design-system-components/lib/components/Entry/EntryCardLoading';
 
 export type EditableReflectionResolverProps = {
   reflectID: string;
   beamID: string;
 };
 
 export const EditableReflectionResolver = (props: EditableReflectionResolverProps) => {
-  const reflectionQuery = useGetReflectionByIdSuspenseQuery({
+  const { getRoutingPlugin } = useRootComponentProps();
+
+  const reflectionReq = useGetReflectionByIdQuery({
     variables: {
       id: props.reflectID,
     },
+    fetchPolicy: 'cache-first',
+    skip: !props.reflectID,
   });
 
-  const { getRoutingPlugin } = useRootComponentProps();
+  if (reflectionReq.loading) return <EntryCardLoading />;
+
+  const entryData =
+    reflectionReq.data?.node && hasOwn(reflectionReq.data.node, 'id')
+      ? reflectionReq.data.node
+      : undefined;
 
   return (
     <React.Suspense>
       <EditableReflection
-        entryData={mapReflectEntryData(
-          hasOwn(reflectionQuery.data?.node, 'id') ? reflectionQuery.data.node : undefined,
-        )}
-        reflectToId={
-          mapReflectEntryData(
-            hasOwn(reflectionQuery.data?.node, 'id') ? reflectionQuery.data.node : undefined,
-          ).id
-        }
+        entryData={mapReflectEntryData(entryData)}
+        reflectToId={mapReflectEntryData(entryData).id}
         contentClickable={true}
         onContentClick={() => {
-          if (hasOwn(reflectionQuery.data?.node, 'id')) {
-            const reflectionId = reflectionQuery.data.node.id;
+          if (hasOwn(reflectionReq.data?.node, 'id')) {
+            const reflectionId = reflectionReq.data.node.id;
             return getRoutingPlugin().navigateTo({
               appName: '@akashaorg/app-akasha-integration',
-              getNavigationUrl: navRoutes => `${navRoutes.Reflect}/${reflectionId}`,
+              getNavigationUrl: navRoutes => `${navRoutes.Reflection}/${reflectionId}`,
             });
           }
         }}
```

Comment:
We can expose this onContentClick as a prop and let the parent handle it. That way we won't need to use navigateTo from the routing plugin but can use the navigate from TanStack router which has TypeScript support.